app/javascript/plain-javascript/quadrant-well-failing.js

```diff
@@ -0,0 +1,47 @@
+document.addEventListener('DOMContentLoaded', () => {
+
+  const failureTable = 'well-failures'
+
+  document.getElementById('quadrant-helper')
+    .querySelectorAll('a[data-select-quadrant-index]')
+    .forEach((button) => {
+      button.addEventListener('click', ()=>{
+        selectQuadrant(button.dataset.selectQuadrantIndex)
+      })
+    })
+
+  const selectQuadrant = (quadrantIndex) => {
+    const checkboxes = document.getElementById(failureTable)
+      .querySelectorAll(`input[data-quadrant-index="${quadrantIndex}"]`)
+    checkboxes.forEach(failWell)
+  }
+
+  const failWell = (checkbox) => {
+    // Failed and cancelled wells can't even be failed in bulk.
+    if (checkbox.dataset.preventQuadrantFail==='true') { return }
+    // Check the box
+    checkbox.checked = true
+    if(checkbox.disabled) {
+      // If the box was disabled it, but record that it was previously disabled,
```

Comment:
If the box was disabled it, ??
If the box was disabled, enable it, ?

Reply:
Thanks, not sure what happened there.